### DISCUSSION TITLE
Managing multiple stored PyPI tokens for multiple projects.

### DISCUSSION THREAD

**Discussion by needs-coffee** [python-poetry/poetry#5620] [Q&A]
The documentation outlines how to store PyPI credentials for publishing packages with `poetry config pypi-token.pypi my-token` or using ENV variables to store the token.

Tokens on PyPI can be created with limited scope to only allow certain projects to use certain tokens, how can you store multiple tokens for multiple different projects in the above manner? 

Is the only current alternative to supply them at command line run or with an ENV variable? (as far as i can ascertain poetry is not able to load variables from `.env` files in the project directory)

thanks!

**Comment by needs-coffee** [comment]
I have had a further look at the source and how passwords are stored - there is currently no mechanism for storing differing authentication tokens /project for the same repository, unless you create a new repository for PyPi with an alias name.

**Reply by abn** [reply]
This is correct. Right now the only way to handle this is as follows.

```sh
poetry config repositories.pypi-project-a https://upload.pypi.org/legacy/
poetry config http-basic.pypi-project-a __token__ <token>
poetry config repositories.pypi-project-b https://upload.pypi.org/legacy/
poetry config http-basic.pypi-project-b __token__ <token>
```

Alaternative as you suggested would be one of the following.
```sh
# use env var
POETRY_PYPI_TOKEN_PYPI=x poetry publish

# use cli
poetry publish -u __token__ -p x
```
